src/tiled/actionshortcuthandler.cpp

```diff
@@ -0,0 +1,143 @@
+void ActionShortcutHandler::addAction(QAction *_newAction)
+{
+    /* Do not add duplicate actions */
+    if (!mActionList.contains(_newAction))
+    {
+        mActionList.append(_newAction);
+        QString str = _newAction->toolTip();
+
+        /* This takes off the existing shortcut text in tooltip if it is instantiated
+         * before being added to this list, this class takes care of the rest  */
+        if (str.contains(QLatin1String("(")) && str.contains(QLatin1String(")")))
+            mTooltipList.append(str.left(str.indexOf(QLatin1String("(")) - 1));
+        else
+            mTooltipList.append(str);
+    }
+}
+
+/* Add all our QActions to a vector, actions are added under these conditions:
+ * 1: the QAction toolTip is not empty || 2: the QAction is not a seperator */
+void ActionShortcutHandler::addActionList(const QVector<QAction*>& listToAdd)
+{
+    foreach(QAction* _action, listToAdd)
+    {
+        // Add each action to our ActionList only if it doesnt already exist
+        if (!mActionList.contains(_action))
+        {
+            if (!_action->toolTip().isEmpty() && !_action->isSeparator())
+            {
+                mActionList.append(_action);
+                QString str = _action->toolTip();
+                /* This takes off the existing shortcut text in tooltip if it is instantiated
+                 * before being added to this list, this class takes care of the rest  */
+                if (str.contains(QLatin1String("(")) && str.contains(QLatin1String(")")))
+                    mTooltipList.append(str.left(str.indexOf(QLatin1String("(")) - 1));
+                else
+                    mTooltipList.append(str);
+            }
+        }
+    }
+}
+
+/* Add all our QWidgets to a vector */
+void ActionShortcutHandler::addDockWidget(QDockWidget* _newWidget)
+{
+    if (!mWidgetList.contains(_newWidget))
+        mWidgetList.append(_newWidget);
+}
+
+/* Add an array of QWidgets to our current List */
+void ActionShortcutHandler::addDockWidgetList(const QVector<QDockWidget*>& listToAdd)
+{
+    foreach(QDockWidget *_widget, listToAdd)
+    {
+        // Add each widget to our action list only if it doesnt already exist
+        if (!mWidgetList.contains(_widget))
+            mWidgetList.append(_widget);
+    }
+}
+
+/* Run through each DockWidget in our widget list, for each DockWidget we need to
+ * grab its toolbar. Then from each toolbar we grab its set of actions. Finally,
+ * we add oll of that toolbars actions to our actionList */
+void ActionShortcutHandler::populateAListFromWList()
+{
+    foreach (QDockWidget* _widget, mWidgetList)
+    {
+        // Find all the childred of type QToolBar of this specific QDockWidget
+        QList<QToolBar*> _toolBars = _widget->findChildren<QToolBar*>();
+
+        // Add the actions of each toolBar in our newly create list of QToolbars
+        foreach (QToolBar* _aToolBar, _toolBars) {
+            addActionList(_aToolBar->actions().toVector());
```

Comment:
Noted, I just went with QVector because the QT documentation stated that QVector should be used over QList. Will change to QList!